src/cluster.c

```diff
@@ -1237,6 +1237,7 @@ void clusterRedirectClient(client *c, clusterNode *n, int hashslot, int error_co
     } else {
         serverPanic("getNodeByQuery() unknown error.");
     }
+    c->duration = 0;
```

Comment:
I think this better follows the fact that the duration is zeroed in rejectCommand and rejectCommandSds. clusterRedirectClient is something similar IMO.

Reply:
> clusterRedirectBlockedClientIfNeeded

We reset duration during the unblock flow in this case, so it should not be needed. 

> clusterRedirectClient is something similar IMO

As it's implemented it is not though. clusterRedirectClient is a specific function for throwing errors while rejectCommand handles a bunch of state related to command rejection during process command. I'd rather modify the reject code so we *also* call some form of rejectClientByRedirect().

Reply:
O.K I agree about changing it back mainly since it indeed introduced a bug (we would have updated command failed+reject stats)
> I'd rather modify the reject code so we also call some form of rejectClientByRedirect().

 I agree that some refactor IS needed around that part. specifically I think we need to consider the blocked commands should follow the logic presented in https://github.com/valkey-io/valkey/commit/90b9f08e5d1657e7bfffe43f31f6663bf469ee75 so that in case blocked commands are also rejected the correct counter will be updated (this should be a different PR)